src/SAWScript/Builtins.hs

```diff
@@ -912,12 +911,12 @@ add_core_defs modname names ss =
       scFindDef sc (qualify n) >>= \maybe_def ->
       case maybe_def of
         Just d -> return d
-        Nothing -> fail $ modname ++ " definition " ++ n ++ " not found"
+        Nothing -> fail $ Text.unpack modname ++ " definition " ++ Text.unpack n ++ " not found"
```

Comment:
Perhaps minimize the number of `unpack` calls here:

```suggestion
        Nothing -> fail $ Text.unpack $ modname <> " definition " <> n <> " not found"
```

Reply:
I was actually thinking about this same change too. (Done in c31fe40b8317d0830926775441d1a651e4ef5261.) It's unfortunate that `fail` uses `String`; I guess that's another reason to try to avoid using `fail`.